service/controller.go

```diff
@@ -1704,7 +1705,8 @@ func (s *service) CreateSnapshot(
 	if req.Name != "" && len(req.Name) > 31 {
 		name := req.Name
 		name = strings.Replace(name, "snapshot-", "sn-", 1)
-		name = name[0:31]
+		length := int(math.Min(float64(len(name)), 31))
```

Comment:
Unfortunately, yes. Golangs `Min` function takes floats as parameters and `len(...)` returns an `int`.

Other approach to avoid using `Min` would be:
```golang
if len(name) > 31 {
    name = name[0:31]
}
```